Firebend.AutoCrud.ChangeTracking.EntityFramework/Abstractions/AbstractChangeTrackingDbContextProvider.cs

```diff
@@ -24,32 +28,67 @@ public abstract class AbstractChangeTrackingDbContextProvider<TEntityKey, TEntit
+            IDbContextConnectionStringProvider<TEntityKey, TEntity> connectionStringProvider,
+            IChangeTrackingOptionsProvider<TEntityKey, TEntity> changeTrackingOptionsProvider)
         {
             _logger = logger;
             _optionsProvider = optionsProvider;
             _connectionStringProvider = connectionStringProvider;
+            _changeTrackingOptionsProvider = changeTrackingOptionsProvider;
+        }
+
+        public async Task<IDbContext> GetDbContextAsync(CancellationToken cancellationToken = default)
+        {
+            var connectionString = await _connectionStringProvider
+                .GetConnectionStringAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            var options = _optionsProvider.GetDbContextOptions(connectionString);
+
+            var context = await GetDbContextAsync(options, cancellationToken);
+
+            return context;
+        }
+
+        public async Task<IDbContext> GetDbContextAsync(DbConnection connection, CancellationToken cancellationToken = default)
+        {
+            var options = _optionsProvider.GetDbContextOptions(connection);
+
+            var context = await GetDbContextAsync(options, cancellationToken);
+
+            return context;
         }
 
         private async Task<IDbContext> GetDbContextAsync(DbContextOptions options, CancellationToken cancellationToken)
         {
-            var context = new ChangeTrackingDbContext<TEntityKey, TEntity>(options);
+            var context = new ChangeTrackingDbContext<TEntityKey, TEntity>(options, _changeTrackingOptionsProvider);
 
             await ChangeTrackingCaches.InitCaches.GetOrAdd(typeof(TEntity).FullName ?? string.Empty, async _ =>
                 {
+                    var type = context.Model.FindEntityType(typeof(ChangeTrackingEntity<TEntityKey, TEntity>));
+                    var schema = type.GetSchema().Coalesce("dbo");
+                    var table = type.GetTableName();
+                    var fullTableName = $"[{schema}].[{table}]";
+
                     try
                     {
                         if (context.Database.GetService<IDatabaseCreator>() is RelationalDatabaseCreator dbCreator)
                         {
-                            await dbCreator
-                                .CreateTablesAsync(cancellationToken)
-                                .ConfigureAwait(false);
+                            var exists = await DoesTableExist(context, schema, table, cancellationToken);
+
+                            if (!exists)
+                            {
+                                await dbCreator
+                                    .CreateTablesAsync(cancellationToken)
+                                    .ConfigureAwait(false);
+                            }
                         }
                     }
                     catch (Exception ex)
```

Comment:
What kinds of exceptions would happen here?  Empty catches are scary.

Reply:
good catch (heh) some things that could occur in this block are the migrations failing, the table exist check failing, etc. i was thinking it would be nice to "push through" any of those issues and still try to log the change further down the stack. in hindsight maybe we should just let it fail or at least logger.log something

Reply:
I don't have a preference.  If it just blows up, it might be easier to find than a logger message.

Reply:
alright i removed the try catch now.